### ISSUE TITLE
NC21 Snap - Whiteboard Integration doesn't has connection issue

### ISSUE THREAD

**Issue by Pilzinsel64** [nextcloud/integration_whiteboard#33]
Hello.

This Whiteboard Integration app doesn't work for me on my Raspberry Pi 4 8GB using Ubuntu Server 21.04. and Nextcloud Snap (NC21, Channel: "latest/beta/pr-1730").
It can be installed but it seems that the process can not be started. When I try to open or create a whiteboard file, the following error comes up.

And just as side node: I upgraded from snap channel latest/stable and I didn't tested it on a fresh installation. I haven't the environment to do this, sorry.

Not formatted:
```
{"reqId":"FycOeiRwSJZSo46tNTbO","level":2,"time":"2021-05-17T12:55:27+00:00","remoteAddr":"5.100.yyy.xxx","user":"zzzzzz","app":"integration_whiteboard","method":"GET","url":"/index.php/apps/integration_whiteboard/space/3330585","message":"Spacedeck request connection error : cURL error 7: Failed to connect to localhost port 9666: Connection refused (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://localhost:9666/api/spaces","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:88.0) Gecko/20100101 Firefox/88.0","version":"21.0.1.1","id":"60a27ff0b5bca"}
```
Formatted:
```
[integration_whiteboard] Warning: Spacedeck request connection error : cURL error 7: Failed to connect to localhost port 9666: Connection refused (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for http://localhost:9666/api/spaces

GET /index.php/apps/integration_whiteboard/space/3330585
from 5.100.yy.xxx by zzzzzz at 2021-05-17T12:55:27+00:00
```

**Comment by NTGDeveloper** [comment]
Same issue. Maybe something to do with ARM? It seems to be common on Raspberry Pi Platforms I think :(

**Comment by Pilzinsel64** [comment]
Could be ... It seems that the arm64 binaries don't get shipped with this app.

**Comment by steffen-moser** [comment]
In my opinion, the Whiteboard Integration App should never ship binaries for a specific platform. Instead, the required framework (e.g. nodejs) should be mentioned in the list of prerequisites, so the server admin can install the necessary packages. For example, we are running Nextcloud on Oracle Solaris 11.4/x64 (SRU 35), so we will also run into this problem, as we cannot execute any Linux binaries.

**Comment by Thesola10** [comment]
I agree with @steffen-moser -- I'm running Nextcloud on NixOS, so obviously the provided Node binary won't work without patches, and I'd much rather just use the system's binary (under `/run/current-system/sw/bin/node`). The [Recognize](https://github.com/marcelklehr/recognize) app has an option to specify the path to Node on the system, so this should have one too.

**Comment by Pilzinsel64** [comment]
Maybe that's the best and easiest solution for cases like this. Well, then it's not an one-click installation how Nextcloud it want's to be, but as optional option e.g. for (yet) unsupported systems or just users that want to user her own binary, it's the best way, I guess.
